x-pack/plugins/reporting/server/export_types/csv_searchsource/generate_csv/generate_csv.ts

```diff
@@ -70,20 +70,35 @@ export class CsvGenerator {
       },
     };
 
-    const results = (
-      await lastValueFrom(this.clients.data.search(searchParams, { strategy: ES_SEARCH_STRATEGY }))
-    ).rawResponse as estypes.SearchResponse<unknown>;
+    let results: estypes.SearchResponse<unknown> | undefined;
+    try {
+      results = (
+        await lastValueFrom(
+          this.clients.data.search(searchParams, { strategy: ES_SEARCH_STRATEGY })
+        )
+      ).rawResponse as estypes.SearchResponse<unknown>;
+    } catch (err) {
```

Comment:
I have found that this try/catch does NOT trigger for request timeouts within the `data.search` method. This can be worked-around by adding an `Rx.timeout` operator call to the observable. However, I don't think that would be the correct solution in this case:
 - we don't have a good value to set for the timeout (testing-wise, there seems to be a hard limit of 30 seconds)
 - a timeout error should be triggered within `data.search` if the waiting time approaches something known to be a problem

cc @ppisljar @Dosant @lukasolson

Reply:
received this reply from @Dosant:

>  I think you don’t get request timeout error because default timeout is 30 seconds, but elasticsearch transport is doing 3 retries. So you’d get a request timeout error in 1 minutes and 30 seconds in this case

> It seems like it is possible [to turn off the retries] with maxRetries: 0 param in es client, but we don’t expose it in search service if you think it is needed, I guess there is no harm in adding it to searchOptions

Reply:
This loose end will be separate work: https://github.com/elastic/kibana/issues/138185